File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l=0,r=0;
        int maxLen=0;
        while(r<nums.length){
            if(nums[r]==0)k--;
            while(k<0){
                if(nums[l]==0) k++;
                l++;
            }
            maxLen=Math.max(maxLen,r-l+1);
            r++;
        }
    return maxLen;
    }
}

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l=0,r=0;
        int res=0;
        while(r<nums.size()){
            if(nums[r]==0) k--;

            while(k<0){
                if(nums[l]==0) k++;
                l++;
            }

            res=max(res,r-l+1);
            r++;
        }
    return res;
    }
};